File: sample-event-emitter/src/test/java/nl/haploid/octowight/AbstractIT.java
package nl.haploid.octowight;

import nl.haploid.octowight.configuration.KafkaConfiguration;
import nl.haploid.octowight.configuration.SamplePostgresConfiguration;
import nl.haploid.octowight.configuration.TestConfiguration;
import nl.haploid.octowight.repository.AtomChangeEventDmoRepository;
import org.junit.Before;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.test.context.ContextConfiguration;
import org.springframework.test.context.junit4.AbstractTransactionalJUnit4SpringContextTests;
import org.springframework.test.context.support.AnnotationConfigContextLoader;

@ContextConfiguration(
		classes = {TestConfiguration.class, KafkaConfiguration.class, SamplePostgresConfiguration.class},
		loader = AnnotationConfigContextLoader.class)
public abstract class AbstractIT extends AbstractTransactionalJUnit4SpringContextTests {

	@Autowired
	private AtomChangeEventDmoRepository repository;

	@Before
	public void setup() {
		repository.deleteAllInBatch();
	}
}

File: event-emitter/src/test/scala/nl/haploid/octowight/emitter/AbstractIT.scala
package nl.haploid.octowight.emitter

import nl.haploid.octowight.emitter.configuration.{TestConfiguration, KafkaConfiguration}
import org.junit.runner.RunWith
import org.scalatest.junit.JUnitRunner
import org.scalatest.tags.Slow
import org.scalatest.{BeforeAndAfterEach, FlatSpec, ShouldMatchers}
import org.springframework.test.context.support.AnnotationConfigContextLoader
import org.springframework.test.context.{ContextConfiguration, TestContextManager}

@RunWith(classOf[JUnitRunner])
@ContextConfiguration(classes = Array(classOf[TestConfiguration], classOf[KafkaConfiguration]),
  loader = classOf[AnnotationConfigContextLoader])
@Slow
abstract class AbstractIT extends FlatSpec with BeforeAndAfterEach with ShouldMatchers {

  override def beforeEach() = new TestContextManager(this.getClass).prepareTestInstance(this)
}
